pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/route/AuthenticationRoute.java

```diff
@@ -80,24 +97,61 @@ public Object handle(Request request, Response response) throws Exception {
+                        () -> {
+                            throw new RuntimeException("Missing " + tenantDomain + " in auth0 claims");
+                        });
+                auth0Util.getClaimValue(auth0Token, clientId, auth0Domain).ifPresentOrElse(claim -> claims.put(clientId, claim.asString()),
+                        () -> {
+                            throw new RuntimeException("Missing " + clientId + " in auth0 claims");
+                        });
+                auth0Util.getClaimValue(auth0Token, userId, auth0Domain).ifPresentOrElse(claim -> claims.put(userId, claim.asString()),
+                        () -> {
+                            throw new RuntimeException("Missing " + userId + " in auth0 claims");
+                        });
 
                 long auth0Expiration = auth0UserInfo.getTokenExpiration();
                 int cookieAgeInSeconds = new Long(auth0Expiration - new Double(System.currentTimeMillis() / 1000d).intValue()).intValue();
 
-                String jwtToken =
-                        new SecurityHelper().createToken(jwtSecret, cookieAgeInSeconds + (System.currentTimeMillis() / 1000) + (60 * 5),
-                                claims);
-
-                DSMToken authResponse = new DSMToken(jwtToken);
-                return authResponse;
-            } else {
-                throw new RuntimeException("UserIdentity not found");
+                String dsmToken = getNewAuth0TokenWithCustomClaims(claims, clientSecret, auth0ClientId, auth0Domain, auth0MgmntAudience, audienceNameSpace);
+                if (dsmToken != null) {
+                    return new DSMToken(dsmToken);
+                } else{
+                    throw new RuntimeException("Not authorized user");
+                }
+            }else{
+                throw new RuntimeException("user was null");
             }
-        } else {
+        }
+        else {
             throw new RuntimeException("There was no token in the payload");
         }
     }
 
+    private String getNewAuth0TokenWithCustomClaims(Map<String, String> claims, String clientSecret, String clientId, String auth0Domain, String auth0Audience, String audienceNameSpace) {
+        String api = "/oauth/token";
+        String contentType = "application/x-www-form-urlencoded";
+        String clientCredentials = "client_credentials";
+
+        String sendRequest = "https://" + auth0Domain + api;
+        Map<String, String> headers = new HashMap<>();
+        headers.put("content-type", contentType);
+
+        Auth0M2MRequest requestModel = new Auth0M2MRequest(clientId, clientCredentials, clientSecret, auth0Audience, claims, audienceNameSpace);
+
+        Auth0M2MResponse response;
+        try {
+            response = DDPRequestUtil.postRequestWithResponse(Auth0M2MResponse.class, sendRequest, requestModel.getParams(), "auth0 M2M", headers);
```

Comment:
postRequestWithResponse as written can return a null. Refactor by either throwing an exception or checking for a null.